From: Vincent Chen <vincent.chen@sifive.com>
Date: Tue, 23 Jun 2020 13:37:25 +0800
Subject: kgdb: Move the extern declaration kgdb_has_hit_break() to generic
 kgdb.h
Git-commit: def0aa218e6d42231540329e6f5741fdec9e7da4
Patch-mainline: v5.8-rc5
References: git-fixes

Currently, only riscv kgdb.c uses the kgdb_has_hit_break() to identify
the kgdb breakpoint. It causes other architectures will encounter the "no
previous prototype" warnings if the compile option has W=1. Moving the
declaration of extern kgdb_has_hit_break() from risc-v kgdb.h to generic
kgdb.h to avoid generating these warnings.

Signed-off-by: Vincent Chen <vincent.chen@sifive.com>
Acked-by: Daniel Thompson <daniel.thompson@linaro.org>
Signed-off-by: Palmer Dabbelt <palmerdabbelt@google.com>
Acked-by: Nikolay Borisov <nik.borisov@suse.com>
---
 include/linux/kgdb.h |    1 +
 1 file changed, 1 insertion(+)

--- a/include/linux/kgdb.h
+++ b/include/linux/kgdb.h
@@ -305,6 +305,7 @@ extern int kgdb_hex2mem(char *buf, char
 
 extern int kgdb_isremovedbreak(unsigned long addr);
 extern void kgdb_schedule_breakpoint(void);
+extern int kgdb_has_hit_break(unsigned long addr);
 
 extern int
 kgdb_handle_exception(int ex_vector, int signo, int err_code,
